core/templates/dev/head/pages/signup/Signup.js

```diff
@@ -143,11 +143,28 @@ oppia.controller('Signup', [
             'Invalid value for email preferences: ' + canReceiveEmailUpdates);
         }
       }
-
       siteAnalyticsService.registerNewSignupEvent();
 
       $scope.submissionInProcess = true;
       $http.post(_SIGNUP_DATA_URL, requestParams).then(function() {
+        var returnUrl = window.decodeURIComponent(
+          urlService.getUrlParams().return_url);
+
+
+        // If the user has the return url as the creator dashboard, it is
```

Comment:
I am facing with finding the return URL. Unfortunately, in the POST request for the signuphandler I do not have the return url part. Should I pass it as a backend parameter? That would be a lot easier I guess. Right now it's very difficult to what I've implemented. What do you think about this method?